.github/workflows/autogenerated-files.yml

```diff
@@ -22,7 +22,7 @@ defaults:
 
 jobs:
   autogenerated:
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
```

Comment:
Most of these should just be `ubuntu-latest`, TBH, since the version doesn't really matter.

Reply:
I'm wondering why these were pinned to a specific ubuntu version?
I can switch to ubuntu-latest, but I'm leaving a little more time for other maintainers to give their opinion on this.

Reply:
A lot of these were because we switched to 22.04 really early - before GitHub had even switched latest.

Reply:
Yea, I guess it would matter if we switched our CI container before GitHub switched `ubuntu-latest`. But we probably shouldn't do that again -- my thinking is that running `brew` on `ubuntu-latest` should always work fine without having to install GCC/glibc/etc.

Last time we were kind of forced to switch early because the ancient image we were using was making building stuff a pain, but I don't think we have that same problem now.

Reply:
I made the change for ubuntu-latest.

> We probably should start taking steps to prepare for the migration though.

I am slowly thinking about starting that project. I just got distracted by the CI using ubuntu 22.04 everywhere so I wanted to change that before making the jump to 24.04

Reply:
> Yea, I guess it would matter if we switched our CI container before GitHub switched `ubuntu-latest`. But we probably shouldn't do that again -- my thinking is that running `brew` on `ubuntu-latest` should always work fine without having to install GCC/glibc/etc.
> 
> Last time we were kind of forced to switch early because the ancient image we were using was making building stuff a pain, but I don't think we have that same problem now.
> 
> We probably _should_ start taking steps to prepare for the migration though.

I agree with all of this. One thing we learned from last time is making forcing a lot of people to use brewed glibc is a really bad idea. We don't test it and there's still some formulae that don't work well with it.

Last time, we ended up breaking things and losing the trust of GitHub. I think we should pay attention to our OS analytics and reflect our choices based on that. Whether we are experiencing pain under the current OS is another important datapoint - what do we gain by bumping the minimum glibc?

In terms of GitHub: also worth noting that `ubuntu-latest` _is_ still 22.04. The 24.04 rollout was cancelled last month due to issues and they haven't tried again yet.